src/scalaDebugger.ts

```diff
@@ -46,15 +50,169 @@ export async function start(
+      return this.askForClassName().then((className) => {
+        if (className === undefined) return undefined;
+        return this.askForArguments().then((args) => {
+          if (args === undefined) return undefined;
+          return this.askForConfigurationName(className).then((name) => {
+            if (name === undefined) return undefined;
+            const result: DebugConfiguration = {
+              type: configurationType,
+              name: name,
+              request: launchRequestType,
+              mainClass: className,
+              buildTarget: buildTarget,
+              args: args,
+            };
+            return result;
+          });
+        });
+      });
+    });
+  }
+
+  private provideDebugTestClassConfiguration(): Thenable<
+    DebugConfiguration | undefined
+  > {
+    return this.askForBuildTarget().then((buildTarget) => {
+      if (buildTarget === undefined) return undefined;
+      return this.askForClassName().then((className) => {
+        if (className === undefined) return undefined;
+        return this.askForConfigurationName(className).then((name) => {
+          if (name === undefined) return undefined;
+          const result: DebugConfiguration = {
+            type: configurationType,
+            name: name,
+            request: launchRequestType,
+            testClass: className,
+            buildTarget: buildTarget,
+          };
+          return result;
+        });
+      });
+    });
+  }
+
+  private askForBuildTarget(): Thenable<string | null | undefined> {
+    return vscode.window
+      .showInputBox({
+        prompt: "Enter name of the build target",
+        placeHolder: "Optional, you can leave it empty",
+      })
+      .then((buildTarget) => {
+        if (buildTarget === undefined) {
+          return undefined;
+        } else if (buildTarget === "") {
+          return null;
+        } else {
+          return buildTarget;
+        }
+      });
+  }
+
+  private askForClassName(): Thenable<string | undefined> {
+    return vscode.window.showInputBox({
+      prompt: "Enter name of the class to debug",
+      placeHolder: "<package>.<Class>",
+    });
+  }
+
+  private askForConfigurationName(
+    className: string
+  ): Thenable<string | undefined> {
+    return vscode.window.showInputBox({
+      prompt: "Enter name of the configuration",
+      value: `Debug ${className}`,
+    });
+  }
+
+  private askForArguments(): Thenable<string[] | undefined> {
+    return vscode.window
+      .showInputBox({ prompt: "Enter argument or leave it empty" })
+      .then((argument) => {
+        if (argument === undefined) {
+          return undefined;
+        } else if (argument === "") {
+          return [];
+        } else {
+          return this.askForArguments().then((rest) => {
+            if (rest === undefined) {
+              return [argument];
+            } else {
+              rest.unshift(argument);
+              return rest;
```

Comment:
I generally avoid mutating arrays, you can use this instead:

```suggestion
              return [argument, ...rest];
```

Reply:
Done. I too had an itch on mutating, but lacked experience with JS.